honeybee_grasshopper_radiance/src/HB Annual Glare Metrics.py

```diff
@@ -0,0 +1,136 @@
+        sGA: Spatial glare autonomy as percentage of area for each analysis grid.
+        pass_fail: A data tree of zeros and ones, which indicate whether a given senor
+            passes the criteria for being free of glare (1) or fails the criteria (0). 
+            Each value is for a different sensor of the grid. These can be plugged
+            into the "LB Spatial Heatmap" component along with meshes of the
+            sensor grids to visualize results.
+"""
+
+ghenv.Component.Name = "HB Annual Glare Metrics"
+ghenv.Component.NickName = 'GlareMetrics'
+ghenv.Component.Message = '1.4.0'
+ghenv.Component.Category = 'HB-Radiance'
+ghenv.Component.SubCategory = '4 :: Results'
+ghenv.Component.AdditionalHelpFromDocStrings = '1'
+
+import os
+
+try:
+    from ladybug.datacollection import BaseCollection
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug:\n\t{}'.format(e))
+
+try:
+    from honeybee_radiance.postprocess.annualglare import glare_autonomy_from_folder
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_radiance:\n\t{}'.format(e))
+
+try:
+    from honeybee_energy.lib.schedules import schedule_by_identifier
+except ImportError as e:  # honeybee schedule library is not available
+    schedule_by_identifier = None
+
+try:
+    from ladybug_rhino.grasshopper import all_required_inputs, list_to_data_tree, \
+        data_tree_to_list
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
+
+
+if all_required_inputs(ghenv.Component):
+    # set default values for the thresholds and the grid filter
+    grid_filter_ = '*' if grid_filter_ is None else grid_filter_
+    _threshold_ = _threshold_ if _threshold_ else 0.4
+
+    # process the schedule
+    if _occ_sch_ is None:
+        schedule = None
+    elif isinstance(_occ_sch_, BaseCollection):
+        schedule = _occ_sch_.values
+    elif isinstance(_occ_sch_, str):
+        if schedule_by_identifier is not None:
+            schedule = schedule_by_identifier(_occ_sch_).values()
+        else:
+            raise ValueError('honeybee-energy must be installed to reference '
+                             'occupancy schedules by identifier.')
+    else:  # assume that it is a honeybee schedule object
+        try:
+            schedule = _occ_sch_.values()
+        except TypeError:  # it's probably a ScheduleFixedInterval
+            schedule = _occ_sch_.values
+    if schedule is not None:
+        bin_schedule = []
+        for val in schedule:
+            bin_val = 1 if val >= 0.1 else 0
+            bin_schedule.append(bin_val)
+        schedule = bin_schedule
+
+    # compute the annual metrics
+    res_folder = os.path.dirname(_results[0]) if os.path.isfile(_results[0]) \
+        else _results[0]
+    GA = glare_autonomy_from_folder(
+        res_folder, schedule, _threshold_, grid_filter_)
+    GA = list_to_data_tree(GA)
+
+    # process the input values into a rokable format
```

Comment:
rockable?

Reply:
Good catch. I guess it's a typo in both components. I imagine it's supposed to read "workable"